quakemigrate/io/data.py

```diff
@@ -271,15 +287,34 @@ def read_waveform_data(self, starttime, endtime, pre_pad=0., post_pad=0.):
                 try:
                     read_start = starttime - pre_pad
                     read_end = endtime + post_pad
-                    st += read(file, starttime=read_start, endtime=read_end,
-                               nearest_sample=True)
+                    st += read(
+                        file,
+                        starttime=read_start,
+                        endtime=read_end,
+                        nearest_sample=True,
+                    )
                 except TypeError:
                     logging.info(f"File not compatible with ObsPy - {file}")
                     continue
 
             # Merge all traces with contiguous data, or overlapping data which
             # exactly matches (== st._cleanup(); i.e. no clobber)
-            st.merge(method=-1)
+            # Apply this on a channel by channel basis so that if any individual
+            # merge fails then only that channel will be omitted.
+            ids = set([tr.id for tr in st])
+            st_merged = Stream()
+            with warnings.catch_warnings():
+                warnings.filterwarnings("error")
+                for tr_id in ids:
+                    try:
+                        st_merged += st.select(id=tr_id).merge(method=-1)
+                    except UserWarning as e:
+                        print(f"\t\t{e}")
+                        print(f"\t\t{st.select(id=tr_id)}")
+                        print(
+                            "\t\tThis channel will not be used for onset "
+                            "calculation."
+                        )
```

Comment:
Is there a reason why these are `print` statements, and not logging?

Reply:
Yeh I also was surprised by this, but it's because it's effectively a standalone function (rather than being called within something that's covered by a QM Run object), so there's no logger spun up to be served these messages

Reply:
No, sorry, I'm being an imbecile. This is not where I thought it was.. fix incoming. I'll try to add a test, too.